### DISCUSSION TITLE
Country code display in Starship terminal

### DISCUSSION THREAD

**Discussion by IOCsec** [starship/starship#4824] [Q&A]
Before Starship I used Powerlevel10k (zsh as Shell), now I use Starship with bash as Shell and **the only thing I miss** is displaying the Country Code based on the IP I am connected to. For example, if I'm using VPN and connected to the US, display the US country code in the terminal, and if I'm connected to a UK server, display the UK country code. 

**Is this possible with Starship?**

See the picture below, how Powerlevel10k displays the country code based on the IP of the VPN you were connected to:

 
![country-code](https://user-images.githubusercontent.com/113497269/212496126-5b61b95d-7dc1-4040-bae2-62abb70db707.png)

**Comment by m19c** [comment]
AFAIK, there is no module for this yet.

**Reply by carboncrystal** [reply]
This feature should be a must.

**Reply by m19c** [reply]
Hmmm... if so, maybe form an issue? I am happy to take it if the primary contributors agree.

**Reply by carboncrystal** [reply]
It would be very useful. I personally agree 100% :smile:

**Reply by chipbuster** [reply]
I wouldn't be opposed. Note that we'd be needing to balance a few competing interests here: we probably don't want to send the user's IP into an online site unless they consent, which would require at least a mention of it in the documentation and a disabled-by-default setup.

Alternately, we could attempt to include a geoIP database within the starship binary itself, but this is likely to have issues with drift as the IPv4 block is fragmented, and might bloat the binary even more than it already is.

**Reply by m19c** [reply]
We should probably talk in general about how to deal with that.

In general, I wouldn't be against the use of external services. However, as you say, it must be a clear opt-in for the user. 

I see another problem: Actually, the provider should only be called if there was a local change of the IP address. Therefore it would be advisable to cache the state in some form. Are there already similar implementations in `starship`?
